x/claims/keeper/integration_test.go

```diff
@@ -0,0 +1,518 @@
+		var actionV int64
+
+		BeforeAll(func() {
+			actionV = actionValue / 3
+			duration := params.DecayStartTime().Sub(s.ctx.BlockHeader().Time)
+			s.CommitAfter(duration)
+			duration = params.GetDurationOfDecay() * 2 / 3
+
+			// create another proposal to vote for
+			testTime := s.ctx.BlockHeader().Time.Add(duration)
+			s.CommitAfter(duration - time.Hour)
+			proposalId = govProposal(priv0)
+			s.CommitAfter(testTime.Sub(s.ctx.BlockHeader().Time))
+		})
+
+		It("can claim ActionDelegate", func() {
+			addr := getAddr(privs[1])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			delegate(privs[1], 1)
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(actionV + int64(prebalance.Amount.Uint64()) - 1)))
+			totalClaimed += actionV
+		})
+
+		It("can claim ActionEVM", func() {
+			addr := getAddr(privs[1])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			sendEthToSelf(privs[1])
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(actionV + int64(prebalance.Amount.Uint64()))))
+			totalClaimed += actionV
+
+			sendEthToSelf(privs[2])
+			totalClaimed += actionV
+		})
+
+		It("can claim ActionVote", func() {
+			addr := getAddr(privs[0])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, params.GetClaimsDenom())
+			govVote(privs[0], proposalId)
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, params.GetClaimsDenom())
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(actionV + int64(prebalance.Amount.Uint64()))))
+			totalClaimed += actionV
+		})
+
+		It("cannot claim ActionDelegate a second time", func() {
+			addr := getAddr(privs[1])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			delegate(privs[1], 1)
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(prebalance.Amount.Uint64() - 1))
+		})
+
+		It("cannot claim ActionEVM a second time", func() {
+			addr := getAddr(privs[1])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			sendEthToSelf(privs[1])
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(prebalance.Amount.Uint64()))
+		})
+
+		It("cannot claim ActionVote a second time", func() {
+			addr := getAddr(privs[0])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, params.GetClaimsDenom())
+			govVote(privs[0], proposalId)
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, params.GetClaimsDenom())
+			Expect(balance.Amount.Uint64()).To(Equal(prebalance.Amount.Uint64()))
+		})
+
+		It("did not clawback to the community pool", func() {
+			// ensure community pool doesn't have the fund
+			poolBalance := s.app.BankKeeper.GetBalance(s.ctx, distrAddr, claimsDenom)
+			Expect(poolBalance.Amount.Uint64()).To(Equal(uint64(0)))
```

Comment:
This was true before the latest claims changes.
Now it fails with:
```
Expected
      <uint64>: 666668
  to equal
      <uint64>: 0
  In [It] at: /Users/loredana/go/src/github.com/tharsis/evmos/x/claims/keeper/integration_test.go:258
```
Is it related to the FeePool?

Reply:
If I replace the check with:
```
poolBalance := s.app.BankKeeper.GetBalance(s.ctx, distrAddr, claimsDenom)
feePool := s.app.DistrKeeper.GetFeePool(s.ctx).CommunityPool.AmountOf(claimsDenom).RoundInt().Uint64()
Expect(poolBalance.Amount.Uint64()).To(Equal(feePool))
```
then it passes. Is this the expected behavior? 
We are in the case where the airdrop is not finished, during the decay duration (after 2/3 decay duration has passed).
@fedekunze , @danburck

Reply:
Solved. With the latest changes, remainders from partial claims (in the decay period) are immediately sent to the community pool when the claim is made: https://github.com/tharsis/evmos/blob/e5075b318020ee4f73d4510a1f2767d2854e055b/x/claims/keeper/claim.go#L121-L128.